misc/dbt-materialize/dbt/adapters/materialize/connections.py

```diff
@@ -47,6 +47,18 @@ def connect(**kwargs):
         # dbt prints notices to stdout, which is very distracting because dbt
         # can establish many new connections during `dbt run`.
         "--welcome_message=off",
+        # Disable warnings about the session's default database or cluster not
+        # existing, as these get quite spammy, especially with multiple threads.
+        #
+        # Details: it's common for the cluster specified in profiles.yml (often
+        # `quickstart`) to be absent. For many dbt deployments, clusters are
```

Comment:
Does this matter? We don't use the cluster specified in `profiles.yml` to create the connection, which is why these notices end up being really annoying. IIRC, passing the target cluster in connections wasn't supported in `libpq`, last we checked!

Reply:
Ah, good point. Updated the comment accordingly.

> passing the target cluster in connections wasn't supported in `libpq`, last we checked!

It is supported in libpq and psycopg2, via `options="--cluster=<NAME>"`—it's the `PostgresConnectionManager` that doesn't make it easy to override just `options` —you have to rip and replace the whole thing, which is a good bit of work.